RecoJets/JetProducers/src/PileupJetIdAlgo.cc

```diff
@@ -582,7 +500,56 @@ PileupJetIdentifier PileupJetIdAlgo::computeIdVariables(const reco::Jet * jet, f
 	internalId_.neuEMfrac_   = pfjet->neutralEmEnergy()    /jet->energy();
 	internalId_.chgHadrfrac_ = pfjet->chargedHadronEnergy()/jet->energy();
 	internalId_.neuHadrfrac_ = pfjet->neutralHadronEnergy()/jet->energy();
-	internalId_.nParticles_ = jet->numberOfDaughters(); 
+	internalId_.nParticles_ = jet->numberOfDaughters();
+
+
+	///////////////////////pull variable///////////////////////////////////
+	float sumW(0.0), sumW2(0.0);
+	float sum_deta(0.0),sum_dphi(0.0);
+	float ave_deta(0.0), ave_dphi(0.0);
+	for (size_t j = 0; j < jet->numberOfDaughters(); j++) {
+	  const auto& part = jet->daughterPtr(j);
+	  if (!(part.isAvailable() && part.isNonnull()) ){
+            continue;
+	  }
+
+	  float weight = part->pt();
+	  float weight2 = weight * weight;
+	  sumW2        += weight2;
+	  float deta = part->eta() - jet->eta();
+	  float dphi = reco::deltaPhi(*part, *jet);
+	  sum_deta     += deta*weight2;
+	  sum_dphi     += dphi*weight2;
+	  if (sumW > 0) {
+	    ave_deta = sum_deta/sumW2;
+	    ave_dphi = sum_dphi/sumW2;
+	  }
+	}
+
+	float ddetaR_sum(0.0), ddphiR_sum(0.0), pull_tmp(0.0);
+	for (size_t i = 0; i < jet->numberOfDaughters(); i++) {
+	  const auto& part = jet->daughterPtr(i);
+	  if (!(part.isAvailable() && part.isNonnull()) ){
+            continue;
+          }
+	  float weight =part->pt()*part->pt();
+	  float deta = part->eta() - jet->eta();
+	  float dphi = reco::deltaPhi(*part, *jet);
+	  float ddeta, ddphi, ddR;
+	  ddeta = deta - ave_deta ;
+	  ddphi = dphi-ave_dphi;
```

Comment:
reco::deltaPhi(dphi, ave_dphi)
unless you are sure this is always small (neither dphi nor ave_dphi goes above pi/2; which is probably correct, considering the constituents are a part of the jet)
[my per-commit comment may go unnoticed]